sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -165,13 +165,14 @@ export class MessageReceiver extends LinkEntity {
   protected _clearAllMessageLockRenewTimers: () => void;
 
   constructor(
-    context: ClientEntityContext,
+    context: ConnectionContext,
+    protected _entityPath: string,
     receiverType: ReceiverType,
     options?: Omit<ReceiveOptions, "maxConcurrentCalls">
   ) {
-    super(context.entityPath, context, {
-      address: context.entityPath,
-      audience: `${context.namespace.config.endpoint}${context.entityPath}`
+    super(_entityPath, context, {
+      address: _entityPath,
+      audience: `${context.config.endpoint}${_entityPath}`
```

Comment:
Would it make more sense to have LinkEntity handle this formatting? Especially now that we're passing in everything it needs?

Reply:
The audience uses different formatting for senders/receivers vs management link. But all 3 extend LinkEntity. So, the parent can't assume the what the audience should look like